muikku-core/src/main/java/fi/otavanopisto/muikku/schooldata/WorkspaceController.java

```diff
@@ -195,16 +195,18 @@ public WorkspaceEntity findWorkspaceEntityByDataSourceAndIdentifier(String schoo
   
   public List<WorkspaceEntity> listWorkspaceEntitiesByUser(UserEntity userEntity, boolean includeUnpublished) {
     List<WorkspaceEntity> result = new ArrayList<>();
-    
     List<WorkspaceUserEntity> workspaceUserEntities = workspaceUserEntityController.listWorkspaceUserEntitiesByUserEntity(userEntity);
     for (WorkspaceUserEntity workspaceUserEntity : workspaceUserEntities) {
-      if (includeUnpublished || workspaceUserEntity.getWorkspaceEntity().getPublished()) {
-        if (!result.contains(workspaceUserEntity.getWorkspaceEntity())) {
-          result.add(workspaceUserEntity.getWorkspaceEntity());
+      WorkspaceEntity workspaceEntity = workspaceUserEntity.getWorkspaceEntity();
+      if (Boolean.TRUE.equals(workspaceEntity.getArchived())) {
```

Comment:
Changing existing methods should be done with care. Filtering archived workspaces here could have unknown effects somewhere else.

Couple of approaches:
* Add another method listUnarchivedWorkspaceEntitiesByUser which basically does the same as the original but filters the archived
* Add a parameter to this method (boolean includeArchived), which if false, filters the archived

Also, it's much more efficient if archived workspaces are filtered in DAO so listWorkspaceUserEntitiesByUserEntity should be replaced with one that handles the filtering while querying the database.